flow-report/Sidebar.tsx

```diff
@@ -0,0 +1,155 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {FunctionComponent} from 'preact';
+import {useMemo} from 'preact/hooks';
+import {useCurrentStep} from './hooks';
+
+export const Hbar: FunctionComponent = () => {
+  return <div className="Hbar"></div>;
+};
+
+export const FlowStepIcon: FunctionComponent<{mode: LH.Gatherer.GatherMode}> = ({mode}) => {
+  return <div className={`FlowStepIcon ${mode}`}></div>;
+};
+
+export const SidebarSummary: FunctionComponent = () => {
+  const currentStep = useCurrentStep();
+  const url = new URL(location.href);
+  url.searchParams.delete('step');
+  return (
+    <a
+      href={url.href}
+      className={`SidebarSummary ${currentStep === null ? 'Sidebar_current' : ''}`}
+      data-testid="SidebarSummary"
+    >
+      <div className="SidebarSummary_icon"></div>
+      <div className="SidebarSummary_label">Summary</div>
+    </a>
+  );
+};
+
+export const SidebarFlowStep: FunctionComponent<{
+  mode: LH.Gatherer.GatherMode,
+  href: string,
+  label: string,
+  hideTopLine: boolean,
+  hideBottomLine: boolean,
+  isCurrent: boolean,
+}> = ({href, label, mode, hideTopLine, hideBottomLine, isCurrent}) => {
+  return (
+    <a
+      className={`SidebarFlowStep ${isCurrent ? 'Sidebar_current' : ''}`}
+      href={href}
+    >
+      <div className="SidebarFlowStep_icon">
+        <div
+          className="SidebarFlowStep_icon_line"
+          style={hideTopLine ? {background: 'transparent'} : undefined}
+        />
+        <FlowStepIcon mode={mode}/>
+        <div
+          className="SidebarFlowStep_icon_line"
+          style={hideBottomLine ? {background: 'transparent'} : undefined}
+        />
+      </div>
+      <div className={`SidebarFlowStep_label ${mode}`}>{label}</div>
+    </a>
+  );
+};
+
+export const SidebarFlow: FunctionComponent<{flowResult: LH.FlowResult}> = ({flowResult}) => {
```

Comment:
`FlowVisualization`? this might be complicated enough to warrant another file with a `sidebar` folder, WDYT?

Reply:
We might have a flow visualization component for the summary page. Another file sounds good.